src/analysis/locally_convex/bounded.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2022 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+import analysis.normed.normed_field
+import analysis.seminorm
+import topology.algebra.module.basic
+import topology.bornology.basic
+
+/-!
+# Von Neumann Boundedness
+
+This file defines von Neumann bounded sets and proves elementary properties.
+
+## Main declarations
+
+* `is_bounded`: A set `s` is bounded if every neighborhood of zero absorbs `s`.
+
+## Main results
+
+* `bounded_bornology`: The set of bounded sets forms a bornology.
+
+## References
+
+* [Bourbaki, *Topological Vector Spaces*][bourbaki1987]
+
+-/
+
+variables {𝕜 E : Type*}
+
+open_locale topological_space pointwise
+
+section semi_normed_ring
+
+section has_zero
+
+variables (𝕜)
+variables [semi_normed_ring 𝕜] [has_scalar 𝕜 E] [has_zero E]
+variables [topological_space E]
+
+/-- A set `s` is bounded if every neighborhood of 0 absorbs `s`. -/
+def is_bounded (s : set E) : Prop := ∀ V ∈ 𝓝 (0 : E), absorbs 𝕜 V s
```

Comment:
I would prefer this (and everything else) to be namespaced. Right now, even `bornology.is_bounded` is namespaced. Aside from being long, I think `bornology.is_von_neumann_bounded` works. Of course, this could be abbreviated to `bornology.is_vn_bounded`, but that abbreviation might do more harm than good. Of course, whatever is done here should be mimicked throughout the file. Perhaps `topological_space` is a better namespace than `bornology` though.

Reply:
There should be two layers of namespaces, right? so we end up with something like `bornology.is_vn_bounded.subset` for instance.

Reply:
Do you think `is_nat_bounded` is ok? I think is `is_natural_bounded` is already long, but it might be still ok.

Reply:
I think `is_nat_bounded` is no good because of confusion with natural numbers; maybe `is_vn_bounded` is better (with an explanation of the abbreivation in the docstring), but someone else should comment on this. I think yes, for two layers of namespaces, but I'm not sure that works with dot notation? I'll have to check.

Reply:
Yes, that is what I suspected as well. We could also move everything into the `topological_space` namespace and then we have `topological_space.is_bounded` and `bornology.is_bounded`.